packages/editor/src/components/local-autosave-monitor/index.js

```diff
@@ -0,0 +1,169 @@
+/**
+ * External dependencies
+ */
+import { once, uniqueId, omit } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useEffect, useRef } from '@wordpress/element';
+import { ifCondition } from '@wordpress/compose';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { parse } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import AutosaveMonitor from '../autosave-monitor';
+
+/**
+ * Time interval at which local autosave should occur after a change, in
+ * seconds.
+ *
+ * @var {number}
+ */
+const AUTOSAVE_INTERVAL_SECONDS = 15;
+
+/**
+ * Function which returns true if the current environment supports browser
+ * localStorage, or false otherwise. The result of this function is cached and
+ * reused in subsequent invocations.
+ */
+const hasLocalStorageSupport = once( () => {
+	try {
+		// Private Browsing in Safari 10 and earlier will throw an error when
+		// attempting to set into localStorage. The test here is intentional in
+		// causing a thrown error as condition bailing from local autosave.
+		window.localStorage.setItem( '__wpEditorTestLocalStorage', '' );
+		window.localStorage.removeItem( '__wpEditorTestLocalStorage' );
+		return true;
+	} catch ( error ) {
+		return false;
+	}
+} );
+
+/**
+ * Custom hook which returns a callback function to be invoked when a local
+ * autosave should occur.
+ *
+ * @return {Function} Callback function.
+ */
+function useAutosaveSaveCallback() {
+	const {
+		postId,
+		getEditedPostAttribute,
+	} = useSelect( ( select ) => ( {
+		postId: select( 'core/editor' ).getCurrentPostId(),
+		getEditedPostAttribute: select( 'core/editor' ).getEditedPostAttribute,
+	} ) );
+
+	return useCallback( () => {
+		window.localStorage.setItem( 'post_' + postId, JSON.stringify( {
```

Comment:
We also should use `sessionStorage`. :)

Reply:
Yes, let's switch to `sessionStorage`-

Reply:
> The REST autosave system has some checks in place to make sure either post_title or content is defined before saving. Worth replicating that here?

Could you point me to what it is you're referring to on this?

Reply:
Yep, it's `isEditedPostSaveable`:
https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/store/selectors.js#L458-L462

Which is used to return early from `savePost` in the event there aren't changes to any of those autosave fields:
https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/store/actions.js#L439

Reply:
I expect the overhead for saving to sessionStorage is very low, possibly lower than retrieving and comparing to the old data?

Reply:
> I expect the overhead for saving to sessionStorage is very low, possibly lower than retrieving and comparing to the old data?

Agreed, unless we measure otherwise.